orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/UpsertIntentTask.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.keel.task
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.KeelService
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.keel.model.Intent
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentDryRunResponse
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentRequest
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentResponse
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import java.util.HashMap
+import java.util.concurrent.TimeUnit
+
+@Component
+class UpsertIntentTask
+@Autowired constructor(
+  private val keelService: KeelService,
+  private val keelObjectMapper: ObjectMapper
+) : RetryableTask {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun execute(stage: Stage) : TaskResult {
+    if (!stage.context.containsKey("intents")) {
+      throw IllegalArgumentException("Missing required task parameter (intents)")
+    }
+
+    if (!stage.context.containsKey("dryRun")) {
+      throw IllegalArgumentException("Missing required task parameter (dryRun)")
+    }
```

Comment:
I would aggregate these required fields together and throw an exception instead. That would allow an end-user to submit once and get all errors without resubmitting and being surprised by another error:

```kotlin
val errors = mutableListOf<String>()
if (!stage.context.containsKey("intents") {
  errors.add("intents")
}
// ...

if (errors.isNotEmpty()) {
  throw IllegalArgumentException("Missing required task parameters (${errors.join(", ")})"
}
```